src/dcd/server/autocomplete/util.d

```diff
@@ -755,14 +755,120 @@ unittest
 
 AutocompleteResponse.Completion makeSymbolCompletionInfo(const DSymbol* symbol, char kind)
 {
-	string definition;
-	if ((kind == CompletionKind.variableName || kind == CompletionKind.memberVariableName) && symbol.type)
-		definition = symbol.type.name ~ ' ' ~ symbol.name;
-	else if (kind == CompletionKind.enumMember)
-		definition = symbol.name; // TODO: add enum value to definition string
+	string definition = symbol.callTip;
+	string symbolName = symbol.name;
+	char completionKind = kind;
+
+	// if symbol has a type, then let's try to resolve it
+	// and properly format the definition
+	if (symbol.type)
+	{
+		DSymbol* sym = cast(DSymbol*) symbol;
+		DSymbol* symType =  sym.type;
+
+		while (symType && sym && sym.type != sym)
+		{
+			// if that's a function, and its type is an alias then stop
+			// otherwise it breaks aliases of local functions
+			if (sym.kind == CompletionKind.functionName && symType.kind == CompletionKind.aliasName)
+				break;
+
+			// otherwise if we resolved something that's not an alias, we can stop
+			else if (sym.kind != CompletionKind.aliasName && symType.kind != CompletionKind.aliasName)
+				break;
+
+			// dummy are internal DCD stuff, as per the documentation on the enum
+			// we can safely ignore it
+			if (symType.kind == CompletionKind.dummy) break;
+
+			if (!symType.type) break;
+
+			sym = symType;
+			symType = symType.type;
+		}
+
+		if (sym)
+		{
+			// if the symbol has a calltip, then let's use that
+			// otherwise build something with its type
+			if (sym.callTip.length > 0 && indexOf(sym.callTip, " ") != -1)
+			{
+				definition = sym.callTip;
+			}
+			else if (sym.type)
+			{
+				string symName = symbol.name;
+				string symTypeName = symType.callTip.length > 0 ? symType.callTip : symType.name;
+				definition = symTypeName ~ " " ~ symName;
+			}
+
+
+			// if that's a function, and definition doesn't have white space
+			// that mean the function returns either auto/enum
+			if (sym.kind == CompletionKind.functionName && indexOf(definition, " ") == -1)
+			{
+				definition = "auto " ~ definition;
+			}
```

Comment:
this whole check might be trying to be too smart, just checking for a space is bound to have issues further down the line e.g. with templates/template arguments that have spaces in them.

I would suggest removing this special case for now to keep the diff smaller, this looks to me rather like something that needs to be changed on dsymbol to be more effective.